autoload/lsp/internal/semantic.vim

```diff
@@ -0,0 +1,216 @@
+let s:use_vim_textprops = lsp#utils#_has_textprops() && !has('nvim')
+let s:use_nvim_highlight = exists('*nvim_buf_add_highlight') && has('nvim')
```

Comment:
We should use `lsp#utils#_has_nvim_buf_highlight()`. Also seems like it is missing `has('nvim')` that we need to add.